src/plugins/query_enhancements/common/types.ts

```diff
@@ -13,3 +14,48 @@ export interface FetchDataFrameContext {
 }
 
 export type FetchFunction<T, P = void> = (params?: P) => Observable<T>;
+
+/**
+ * Job states are defined to mostly be compatible with EMR job run states, but aren't strictly
+ * limited to that use. "PENDING" has been replaced with "WAITING", since the underlying async API
+ * seems to use this state instead. See also: {@link parseJobState}.
+ */
+export enum JobState {
+  SUBMITTED = 'SUBMITTED',
+  WAITING = 'WAITING',
+  SCHEDULED = 'SCHEDULED',
+  RUNNING = 'RUNNING',
+  FAILED = 'FAILED',
+  SUCCESS = 'SUCCESS',
+  CANCELLING = 'CANCELLING',
```

Comment:
Nit: We'll probably centralize all of these types post 2.16, think there's 3 locations now where this exists. Aside from that, is `CANCELLING` something new from the backend API? Don't remember this being in the old `DirectQueryLoadingStatus` type.

Reply:
I ported all the [EMR job run states](https://docs.aws.amazon.com/emr/latest/EMR-Serverless-UserGuide/job-states.html), with one of them changed since the flint API seems to use a different term. I'm not sure what all of the states we actually use are.

Reply:
definitely should be one. incase the job is just too long and we cant abort it from the front end